assignments/array-methods.js

```diff
@@ -53,31 +53,164 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
+logHeader('CHALLENGE 4: Use .reduce()', 'Add up all the donations into a ticketPriceTotal array and log the result:');
+
+/* 
+    The donations need to be tallied up and reported for tax purposes. Add up all 
+    the donations into a ticketPriceTotal array and log the result.
+*/
+
+let ticketPriceTotal = runners.map(runner => runner.donation).reduce((dono1, dono2) => dono1 + dono2);
+
 console.log(ticketPriceTotal);
 
-// ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+
+
+
+/*********************************************************************************
+*                             CHALLENGE 5: Be Creative                           *
+**********************************************************************************/ 
+logHeader('CHALLENGE 5: Be Creative', 'Try to create and then solve 3 unique problems using one or many of the array methods');
+/*
+    ==== Challenge 5: Be Creative ====
+    Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  
+    Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+*/
 
 // Problem 1
 
+console.log('\nJuan Berzon and Charleen Sheering just got married. They\'re a very progressive, forward-thinking couple, so update both their lastname\'s to Charleen\'s last name.\n');
+
+let married = runners.filter(runner => (runner.first_name === 'Juan' && runner.last_name === 'Berzon') || (runner.first_name === 'Charleen' && runner.last_name === 'Sheering'))
+                     .map(runner => {
+                         runner.last_name = 'Sheering';
+                         return runner;
+                     });
+
+console.log(married);
+
 // Problem 2
 
-// Problem 3
\ No newline at end of file
+console.log('\n\nThe event director has decided to match 5% of the total donations of all runners and to add that amount to each runners donation. He\'s filthy rich.');
+
+let getDonations = runner => `${runner.id}: dono = ${runner.donation}`;
+
+let runnerDonationsBefore = runners.map(getDonations);
+
+let totalDonations = runners.map(runner => runner.donation).reduce((dono1, dono2) => dono1 + dono2);
+
+let directorDonationMatch = Math.round(totalDonations * .05);
+
+runners.forEach(runner => {
+    runner.donation += directorDonationMatch;
+});
+
+let runnerDonationsAfter = runners.map(getDonations);
+
+console.log('runner donations before:');
+console.log(runnerDonationsBefore);
+
+console.log(`\nDirectors Donation Match: ${directorDonationMatch}\n\n`);
+console.log('runner donations after:');
+console.log(runnerDonationsAfter);
+
+// Problem 3
```

Comment:
Uh oh, you forgot problem 3! Remember, this isn't stretch, so don't forget...

...you know what? I'll take your "consoleCannon" and we'll call it even. How about that? 😂